### PR TITLE
Stack config doc

### LINKED ISSUE #272: Update Kabanero CR instance configuration for Stacks / Versions
Both the conversion from collections to stacks, and the multi-version stack support, make updates necessary to https://kabanero.io/docs/ref/general/configuration/kabanero-cr-config.html.

We need to wholesale compare the Kabanero CRD to what's here, and make necessary changes.  Similarly, the Collection is replaced with Stack, so the Stack CRD should be compared against what is here now for Collection and updates should be made.

### PR DESCRIPTION
**Pull request by kaczyns**
Fixes #272 
This contains most (if not all) of the updates to the Kabanero and Stack CR instances for Kabanero 0.6.0.  Additions include:
* Stack CR instance (new)
* CodeReady Workspaces configuration
* RH SSO configuration

Note that this is not complete documentation for either CodeReady or RH SSO - this is just documenting what can be configured in the Kabanero and Stack CR instances.

### CODE REVIEW

File: ref/general/configuration/kabanero-cr-config.adoc [hunk 1]
```diff
@@ -110,6 +110,14 @@ specify the following fields:
...
 ***** `tlsSupport` - Controls whether TLS is enabled for the CodeReady
       Workspaces instance.  Valid values are `true` and `false`.
+** `sso` - Configuration for the Red Hat SSO Server
+*** `enable` - Controls whether Kabanero will deploy an instance of 
+    the Red Hat SSO Server for application use.  Valid values are `true` a
+    nd `false`.  The default value is `false`.
+*** `adminSecretName` - The name of a secret that contains keys for
+    `username`, `password`, and `realm`.  These values will become the
+    admin username and password for the Red Hat SSO server, as well as
+    define the realm that the SSO server will use.
```
Comment (meiaus) [inline ref/general/configuration/kabanero-cr-config.adoc]: Can we match the key names referred here to what are created in the prereq page: https://github.com/kabanero-io/docs/pull/269 (after the latest update in 269)?

Can we add one quick line to hint that the secret should have been created from the prereq page (a link) when setting enable: true?

File: ref/general/configuration/kabanero-cr-config.adoc [hunk 2]
```diff
@@ -6,16 +6,16 @@
 = Configuring a Kabanero CR instance
 
 A `Kabanero` custom resource (CR) instance describes a particular Kabanero
-installation.  This description includes the locations of the Kabanero
-Collections, and how the Kabanero CLI should interact with GitHub.
+installation.  This description includes the locations of the Appsody
```
Comment (jantley-ibm) [inline ref/general/configuration/kabanero-cr-config.adoc]: Change
Appsody Stacks
to
application stacks

File: ref/general/configuration/kabanero-cr-config.adoc [hunk 3]
```diff
@@ -25,41 +25,60 @@ specify the following fields:
 ** `version` - The release of Kabanero that you want to install.
    The levels of the individual components, such as the Appsody operator
```
Comment (jantley-ibm) [inline ref/general/configuration/kabanero-cr-config.adoc]: Change
Appsody operator
to
application operator

File: ref/general/configuration/kabanero-cr-config.adoc [hunk 4]
```diff
@@ -25,41 +25,60 @@ specify the following fields:
...
    and CLI services, are selected automatically based on this level.  The
-   release follows semver notation (for example, 0.2.0).
+   release follows semver notation (for example, 0.6.0).
 ** `targetNamespaces` - A list of namespaces which the Appsody operator will
```
Comment (jantley-ibm) [inline ref/general/configuration/kabanero-cr-config.adoc]: Change
Appsody operator
to
application operator

File: ref/general/configuration/kabanero-cr-config.adoc [hunk 5]
```diff
@@ -87,40 +129,36 @@ Kabanero instance:
...
-    looking in the `errorMessage` attribute.
-*** `errorMessage` - Provides more details for a `ready` status of `False`.
+    looking in the `message` attribute.
+**** `message` - Provides more details for a `ready` status of `False`.
     The error message is copied from the `ready` condition on the
     `KnativeServing` CR instance.
-*** `version` - The version of Knative Serving as reported by the
+**** `version` - The version of Knative Serving as reported by the
     `KnativeServing` CR instance.
 ** `tekton` - Contains information about the Tekton instance which was found
```
Comment (jantley-ibm) [inline ref/general/configuration/kabanero-cr-config.adoc]: Change
Tekton instance
to
pipelines instance

File: ref/general/configuration/kabanero-cr-config.adoc [hunk 6]
```diff
@@ -87,40 +129,36 @@ Kabanero instance:
...
    on this cluster.
 *** `ready` - The overall status of Tekton, as reported by the
```
Comment (jantley-ibm) [inline ref/general/configuration/kabanero-cr-config.adoc]: Change both instances of 
Tekton
to
pipelines

File: ref/general/configuration/kabanero-cr-config.adoc [hunk 7]
```diff
@@ -129,59 +167,73 @@ Kabanero instance:
...
 ** `landing` - Contains information about the Kabanero landing page
 *** `ready` - The overall status of the Kabanero landing page.
     A value of `True` indicates the landing page was deployed successfully.
     A value of `False` indicates there was a problem, and more information can
-    be found by looking in the `errorMessage` attribute.
-*** `errorMessage` - Provides more details for a `ready` status of `False`.
+    be found by looking in the `message` attribute.
+*** `message` - Provides more details for a `ready` status of `False`.
 *** `version` - The version of the landing page that was deployed.
 ** `appsody` - Contains information about the Appsody operator that was
```
Comment (jantley-ibm) [inline ref/general/configuration/kabanero-cr-config.adoc]: change
Appsody operator
to
application operator

File: ref/general/configuration/kabanero-cr-config.adoc [hunk 8]
```diff
@@ -129,59 +167,73 @@ Kabanero instance:
...
    deployed by the Kabanero operator.
 *** `ready` - The overall status of the Appsody operator.  A value
```
Comment (jantley-ibm) [inline ref/general/configuration/kabanero-cr-config.adoc]: change
Appsody operator
to
application operator

File: ref/general/configuration/stack-cr-config.adoc [hunk 9]
```diff
@@ -0,0 +1,86 @@
+:page-layout: doc
+:page-doc-category: Configuration
+:page-title: Configuring a Stack CR Instance
+:sectanchors:
+= Configuring a Stack CR instance
+
+A `Stack` custom resource (CR) instance describes a particular Appsody
```
Comment (jantley-ibm) [inline ref/general/configuration/stack-cr-config.adoc]: change
Appsody stack
to
application stack

File: ref/general/configuration/stack-cr-config.adoc [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+       validate SSL/TLS certificates presented by the pipelines URL.
+       Valid values are `true` and `false`.
+*** `images` - A list of container images that are associated with
+    this version of the stack.  A `PipelineRun` may include a step
+    that validates that an application is build using an image that
+    is included in this list.
+**** `id` - A descriptive name of this image
+**** `image` - The location of this image in the container registry.
+
+== Inspecting your Stack CR Instances
```
Comment (jantley-ibm) [inline ref/general/configuration/stack-cr-config.adoc]: change
Instances
to
instances

File: ref/general/configuration/stack-cr-config.adoc [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+command:
+
+`oc get stacks -n kabanero -o yaml`
+
+The example uses the kabanero namespace.  To use a different namespace,
+replace `-n kabanero` with the name of another namespace.
+
+== Examples
+
+The following yaml defines a `Stack` instance with a single version of
```
Comment (jantley-ibm) [inline ref/general/configuration/stack-cr-config.adoc]: Change
The following
to
This example

File: ref/general/configuration/kabanero-cr-config.adoc [hunk 12]
```diff
@@ -195,45 +247,45 @@ replace `-n kabanero` with the name of another namespace.
 
 == Examples
 
-The following yaml defines a `Kabanero` instance at version 0.2.0, using
-the default collection set.
+The following yaml defines a `Kabanero` instance at version 0.6.0, using
```
Comment (jantley-ibm) [inline ref/general/configuration/kabanero-cr-config.adoc]: Change
The following
to
This example

File: ref/general/configuration/kabanero-cr-config.adoc (hunk 1 above)
Comment (kaczyns) [reply ref/general/configuration/kabanero-cr-config.adoc]: Hi - I can link this page to the prereq page (#269).  If you need additional validation documentation, you will need to add that information somewhere else.  I don't understand how to validate the SSO installation.

Comment (kaczyns) [reply ref/general/configuration/kabanero-cr-config.adoc]: I spoke with @brutif and the key names in the sso page are wrong.  We'll fix with a PR.

**Comment by kaczyns** [comment]
@jantley-ibm I think I got all of the terminology changes.  Please give it another look over and let me know.  Thanks!

File: ref/general/configuration/kabanero-cr-config.adoc (hunk 1 above)
Comment (meiaus) [reply ref/general/configuration/kabanero-cr-config.adoc]: thanks.  will check again from sso page for validation.

File: ref/general/configuration/kabanero-cr-config.adoc [hunk 13]
```diff
@@ -70,12 +89,36 @@ specify the following fields:
 *** `enable` - Controls whether the Kabanero landing page is deployed by
     the Kabanero operator.  Valid values are `true` and `false`.  The default
     value is `true`.
-** `che`
-*** `enable` - Controls whether Kabanero will deploy an instance of Che.
-    Valid values are `true` and `false`.  The default value is `false`.
-*** `cheOperatorInstance`
-**** `cheWorkspaceClusterRole` - The workspace cluster role used
-     by Che.  By default, the `eclipse-codewind` cluster role is used.
+** `codereadyWorkspaces`
```
Comment (johnmcollier) [inline ref/general/configuration/kabanero-cr-config.adoc]: This should be `codeReadyWorkspaces`. See https://github.com/kabanero-io/kabanero-operator/blob/master/config/samples/full.yaml#L72

File: ref/general/configuration/kabanero-cr-config.adoc [hunk 14]
```diff
@@ -87,101 +130,111 @@ Kabanero instance:
...
 *** `uiLocations` - The location (URL) of the UI endpoint of kAppNav.
     This information is copied from the `Route` for the kAppNav UI service.
 *** `apiLocations` - The location (URL) of the API endpoint of
     kAppNav.  This information is copied from the `Route` for the kAppNav API
     service.
-** `che` - Contains information about the Che instance that is deployed by
-   the Kabanero operator.
-*** `ready` - The overall status of Che.  A value of `True`
-    indicates Che was installed and configured successfully.  A value of
+** `codereadyWorkspaces` - Contains information about the CodeReady
```
Comment (johnmcollier) [inline ref/general/configuration/kabanero-cr-config.adoc]: This should be `codeReadyWorkspaces` See https://github.com/kabanero-io/kabanero-operator/blob/master/config/samples/full.yaml#L72

**Pull request merged**